packages/example-app-base/src/utilities/SiteDefinition.types.ts

```diff
@@ -1,8 +1,34 @@
-import { ICustomizations } from 'office-ui-fabric-react';
+import * as React from 'react';
+import { ICustomizations, IContextualMenuItem } from 'office-ui-fabric-react';
 import { INavPage } from '../components/Nav/index';
 import { IPlatform } from '../components/PlatformPicker/index';
 import { ISiteMessageBarProps } from '../components/SiteMessageBar/index';
 
+/**
+ * Interface defining the necessary information to populate the version switcher.
+ */
+export interface IVersionSwitcherDefinition {
+  /**
+   * The list of available versions whose documentation is presented on the website.
+   */
+  versions?: string[];
```

Comment:
Shouldn't there always be at least 1 version? (IOW, make this required instead of optional)

Reply:
Theoretically the SiteDefinition type (and the rest of the site infrastructure) could be used by another app which doesn't have this notion. Practically we don't expect that to happen, but that's the theoretical reason for not adding required properties.